### ISSUE TITLE
imageCompletion called twice for capturePictureWithCompletion with .writeFilesToPhoneLibrary = true

### ISSUE THREAD

**Issue by franckclement** [imaginary-cloud/CameraManager#187]
Hey,

It turns out that when you set

```swift
cameraManager.writeFilesToPhoneLibrary = true
```

and then call 

```swift
cameraManager.capturePictureWithCompletion {}
```

The ```imageCompletion``` closure get called twice.

Here is a workaround t'ill it's fixed

```swift
cameraManager.capturePictureWithCompletion { [weak self] result in
   guard let `self` = self else { return }
   switch result {
      case .success(let content):
        switch content {
          case .image(let photo):
            // update your layout ....
          default: break
         }
       case .failure(let error):
          // Handle error
    }
}
```

switching over ```content``` and using ```.image``` case let ou get notified once per image captured, but if you need to use the ```.asset``` case then you'll get notified twice.

**Comment by fenixsolorzano** [comment]
This happens to me too

**Comment by Johnson145** [comment]
Right. As a part of https://github.com/imaginary-cloud/CameraManager/commit/8d3a8a7a9e39167f65c374bb3c183cbd447667fe the following lines were added: https://github.com/imaginary-cloud/CameraManager/blob/05a66bcfdae725029c60772f0cb4a51e04877340/camera/CameraManager.swift#L643-L649

I guess we shouldn't trigger the `imageCompletion` callback there, right? My suggestions: We could either spend some time to include a success/failure information about the optional library storage to the primary callback or simply drop this part and replace it by a simple log line?

**Comment by franckclement** [comment]
@Johnson145 Yeah, I guess a simple log line should be OK. 

At the end of this function ([line 577](https://github.com/imaginary-cloud/CameraManager/blob/05a66bcfdae725029c60772f0cb4a51e04877340/camera/CameraManager.swift#L577)):

```swift
_capturePicture(_ imageData: Data, _ imageCompletion: @escaping (CaptureResult) -> Void)
```
 There is a callback to `imageCompletion` escaping closure, and the failure when the image is saved to library is caught, so callbacks seems consistent and not called twice with this little update 👍🏻

**Comment by Johnson145** [comment]
That's right. However, I wouldn't keep only that outer callback, because it will get triggered before the optional library storing finished (whether successful or not). That's why I kept the inner callbacks in my pull request. Instead I prevent the outer callback with a simple else clause. The latter is actually exactly the same way which is already applied to video recording.

One may still argue whether it's a good idea that a failure during the library storing should imply a total failure or not. For now, I've just added a short TODO comment in the PR.

**Comment by tapannathvani** [comment]
I am facing the same issue. Even I followed @franckclement work around.

@objc func captureImage(){
        cameraManager.capturePictureWithCompletion({ [weak self] result in
            guard let `self` = self else { return }

            switch result {
            case .failure:
                self.cameraManager.showErrorBlock("Error occurred", "Cannot save picture.")
            case .success(let content):
                switch content {
                case .image(let photo):
                    self.thumbnailImage.image = photo
                default: break
                }
            }
        })
    }

**Comment by torrao** [comment]
@franckclement @fenixsolorzano @Johnson145 @tapannathvani 

just released v5.1.0 where this should be fixed.
